### PR TITLE
Adding documentation for multi-device in Tensorflow

### PR DESCRIPTION
**Pull request by kavyasrinet**
Adding a document to explain how multi-device works in Tensorflow.

### CODE REVIEW

File: doc/design/multi_gpu_tf.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+with tf.device('/cpu:0'):
+  sum = tf.add_n(c)
+
+sess = tf.Session(config=tf.ConfigProto(log_device_placement=True))
+
+print(sess.run(sum))
+```
+Here we are running the same operation(MatMul) on two different GPUs : '/gpu:2' and '/gpu:3', and on the CPU ('/cpu:0') the operation `tf.add_n` which adds all input tensors element-wise. This operation would collect the output of MatMulfrom the GPUs and aggregate them on the CPU.
+
+The [tutorial](https://www.TensorFlow.org/tutorials/deep_cnn) on CIFAR10 is a good example demonstrating how to do training with multiple GPUs on TensorFlow.
```
Comment (wangkuiyi) [inline doc/design/multi_gpu_tf.md]: We might need more information about Device Contexts and how they are used to build new operators/kernels?

**Comment by kavyasrinet** [comment]
Thank you for the review. I think Wiki page is a good idea, since it is meant for our educational purpose anyway. I will create a page and close this PR.

File: doc/design/multi_gpu_tf.md (hunk 1 above)
Comment (kavyasrinet) [reply doc/design/multi_gpu_tf.md]: That's a good suggestion. Will dig up more on this and add a section.

**Comment by kavyasrinet** [comment]
Closing this PR and created the Wiki page: https://github.com/PaddlePaddle/Paddle/wiki/Understanding-Multi-device-training-in-TensorFlow as per @wangkuiyi 's review.

**Pull request closed without merging**